modules/compute/src/main/java/org/apache/ignite/internal/compute/ComputeUtils.java

```diff
@@ -509,7 +510,7 @@ static ComputeJobDataHolder marshalAndWrapResult(Object result, @Nullable Marsha
             // Value is represented by 3 tuple elements: type, scale, value.
             var builder = new BinaryTupleBuilder(3, 3, false);
             ClientBinaryTupleUtils.appendObject(builder, result);
-            return new ComputeJobDataHolder(NATIVE, builder.build().array());
+            return new ComputeJobDataHolder(NATIVE, IgniteUtils.byteBufferToByteArray(builder.build()));
```

Comment:
Unnecessary copy, we can refactor `ComputeJobDataHolder` to hold `ByteBuffer` instead of `byte[]`.

Reply:
This is always a heap buffer, as I understand.

Also, is there something like `writeReplace` in network messaging?